tests/cpp/test_gpu_view.cpp

```diff
@@ -2619,4 +2620,140 @@ TEST_F(GpuViewTest, GroupNorm) {
   testValidate(
       executor_cache.fusion(), cg_outputs, {t0}, {ref}, __LINE__, __FILE__);
 }
+
+TEST_F(GpuViewTest, FusionMismatchingReshape) {
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  // TODO: use symbolic sizes. Currently, this is not working because of
+  // failures in replaceSymbolicSizes
+  TensorView* tv0 = makeContigConcreteTensor({2, 3, 5}, DataType::Float);
+  // TensorView* tv0 = makeContigTensor(3, DataType::Float);
+  fusion.addInput(tv0);
+  auto tv1 = sin(tv0);
+  auto tv2 = reshape(tv1, {2, 3, 5}, {2 * 3, 5});
+  auto tv3 = reshape(tv1, {2, 3, 5}, {2, 3 * 5});
+  auto tv4 = cos(tv2);
+  auto tv5 = exp(tv3);
+  fusion.addOutput(tv4);
+  fusion.addOutput(tv5);
+
+  // For this fusion, because tv4 and tv5 have different views, and these views
+  // are not compatible, traditionally, we were not able to support this fusion.
+  // However, with the advanced feature that domains in a TensorDomain can be
+  // connected by both forward and backward, we can schedule the entire fusion
+  // like tv4 as follows:
+
+  // First, the most difficult part is to schedule tv5. The TensorDomain of
+  // tv5 only contains 2 IDs, and non of them is mapped to anything in tv4.
+  // So, in order to schedule tv5 like tv4, we need to reconstruct the entire ID
+  // graph into tv5.
+
+  // Before schedule, tv5 is:
+  //
+  //  logical domain: [I0, I1]
+  //
+  // Now, we want to make tv5 as:
+  //
+  //                    I2   I3
+  //                      \ /
+  //  logical domain: [I0, I1]
```

Comment:
Why is this necessary? Doesn't the exact graph already have this connection?

Reply:
Without this, how do you schedule the loop domain of tv5?

Reply:
Ah, but we still need to create an actual transformation, so that's why we want to have this expr for tv5?

Reply:
> Without this, how do you schedule the loop domain of tv5?

For some reason, I was thinking about something like using the loop domain of tv4 directly. I think we chatted a little about this line of ideas like using AbstractTensor to represent loop domains without actually scheduling each tensor.

As an incremental step, this PR makes a lot more sense.

Reply:
> Ah, but we still need to create an actual transformation, so that's why we want to have this expr for tv5?

Yes

Reply:
> For some reason, I was thinking about something like using the loop domain of tv4 directly.

Yes, I believe this is what we will do in the long term, because this is the only way I can think of automating the schedule of reshape. But in the short term, as you pointed out, making this work is a good incremental step to take.